docs/rules/prefer-user-event.md

```diff
@@ -0,0 +1,126 @@
+# Use [userEvent](https://github.com/testing-library/user-event) over using `fireEvent` for user interactions (prefer-user-event)
+
+From
+[testing-library/dom-testing-library#107](https://github.com/testing-library/dom-testing-library/issues/107):
+
+> [...] it is becoming apparent the need to express user actions on a web page
+> using a higher-level abstraction than `fireEvent`
+
+`userEvent` adds related event calls from browsers to make tests more realistic than its counterpart `fireEvent`, which is a low-level api.  
+See the appendix at the end to check how are the events from `fireEvent` mapped to `userEvent`
+
+## Rule Details
+
+This rules enforces the usage of [userEvent](https://github.com/testing-library/user-event) methods over `fireEvent`. By default, the methods from `userEvent` takes precedence, but you add exceptions by configuring the rule in `.eslintrc`
+
+See below for examples of valid usages of `fireEvent` methods with the configuration.
+
+Examples of **incorrect** code for this rule:
+
+```ts
+// a method in fireEvent that has a userEvent equivalent
+import { fireEvent } from '@testing-library/dom';
+fireEvent.click(node);
+
+// using fireEvent with an alias
+import { fireEvent as fireEventAliased } from '@testing-library/dom';
+fireEventAliased.click(node);
+
+// using fireEvent after importing the entire library
+import * as dom from '@testing-library/dom';
+dom.fireEvent.click(node);
+```
+
+Examples of **correct** code for this rule:
+
+```ts
+// any userEvent method
+userEvent.click();
+// fireEvent method that does not have an alternative in userEvent
```

Comment:
Sorry, I feel like a jerk to nit about the style, but since this is public I think it should look right 😅

```suggestion

// fireEvent method that does not have an alternative in userEvent
```